### ISSUE TITLE
Reward

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/trl#2633]

**Comment by qgallouedec** [comment]
```python
from datasets import load_dataset
from transformers import AutoTokenizer, AutoModelForSequenceClassification

from trl import RewardConfig, RewardTrainer

model_id = "trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5"
model = AutoModelForSequenceClassification.from_pretrained(model_id)
tokenizer = AutoTokenizer.from_pretrained(model_id)
model.config.pad_token_id = tokenizer.eos_token_id

dummy_dataset = load_dataset("trl-lib/hh-rlhf-helpful-base", split="train[:100]")

batch_size = 8
gradient_accumulation_steps = 1
output_dir = f"ORM-bsz{batch_size}-grad_acc{gradient_accumulation_steps}"

training_args = RewardConfig(
    output_dir=output_dir,
    per_device_train_batch_size=batch_size,
    gradient_accumulation_steps=gradient_accumulation_steps,
    logging_steps=2,
)
trainer = RewardTrainer(model=model, args=training_args, processing_class=tokenizer, train_dataset=dummy_dataset)

trainer.train()
```

![Image](https://github.com/user-attachments/assets/7895a65f-81dc-4263-915a-174fc303a827)


Currently I don't know why we don't have the issue here.
